### PR TITLE
Fixes #610: Using LogMessageKeys everywhere

### LINKED ISSUE #610: Logging should use LogMessageKeys consistently
- Currently, logging code that uses KeyValueLogMessages, uses LogMessageKeys inconsistently.
- Some instances use LogMessageKeys, while others use arbitrary strings.
- All logging should consistently use LogMessageKeys.

### PR DESCRIPTION
**Pull request by vsharma999**
- Added new keys to LogMessageKeys.
- Fixed all KeyValueLogMessage.of invocations to use keys from LogMessageKeys.
- When two differing keys used the same name, used '_C' suffix to denote
  camel-case.

### CODE REVIEW

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java [hunk 1]
```diff
@@ -32,14 +32,28 @@
 public enum LogMessageKeys {
     // general keys
     TITLE("ttl"),
+    CLUSTER("cluster"),
     SUBSPACE("subspace"),
     SUBSPACE_KEY("subspace_key"),
+    INDEX_SUBSPACE("indexSubspace"),
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: We weren't very consistent in our code, using camel-case in some places and underscores in another.  I think for consistency, we should use this as an opportunity to be consistent, so this should be `index_subspace` and, similarly the other camel-case's here should be updated.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java [hunk 2]
```diff
@@ -349,7 +350,8 @@ private RunRetriable(@Nullable List<Object> additionalLogMessageKeyValues) {
                 if (tries + 1 < getMaxAttempts() && retry) {
                     if (LOGGER.isWarnEnabled()) {
                         final KeyValueLogMessage message = KeyValueLogMessage.build("Retrying FDB Exception",
-                                "message", fdbMessage, "code", code);
+                                                                LogMessageKeys.MESSAGE      , fdbMessage,
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java]: As a rule we never try to column-align values like this as it makes editing that much more difficult later. So this should just be:

    LogMessageKeys.MESSAGE, fdbMessage,
    LogMessageKeys.CODE, code

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java [hunk 3]
```diff
@@ -260,9 +260,9 @@ public void setMaintainHistory(boolean maintainHistory) {
                 }
                 if (LOGGER.isDebugEnabled()) {
                     LOGGER.debug(KeyValueLogMessage.of("Cached serialized meta-data is out-of-date",
-                            subspaceProvider.logKey(), subspaceProvider.toString(context),
-                            LogMessageKeys.VERSION, currentVersion,
-                            "cachedVersion", cachedSerializedVersion));
+                                    subspaceProvider.logKey()     , subspaceProvider.toString(context),
+                                    LogMessageKeys.VERSION        , currentVersion,
+                                    LogMessageKeys.CACHED_VERSION , cachedSerializedVersion));
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java]: Remove column alignment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java [hunk 4]
```diff
@@ -380,9 +380,9 @@ protected RecordMetaData buildMetaData(@Nonnull RecordMetaDataProto.MetaData met
                 int oldVersion = oldProto.getVersion();
                 if (metaDataProto.getVersion() <= oldVersion) {
                     LOGGER.warn(KeyValueLogMessage.of("Meta-data version did not increase",
-                            subspaceProvider.logKey(), subspaceProvider.toString(context),
-                            "old", oldVersion,
-                            "new", metaDataProto.getVersion()));
+                                    subspaceProvider.logKey() , subspaceProvider.toString(context),
+                                    LogMessageKeys.OLD        , oldVersion,
+                                    LogMessageKeys.NEW        , metaDataProto.getVersion()));
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java]: Remove column alignment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java [hunk 5]
```diff
@@ -439,9 +439,9 @@ protected RecordMetaData buildMetaData(@Nonnull RecordMetaDataProto.MetaData met
                             }
                             if (LOGGER.isDebugEnabled()) {
                                 LOGGER.debug(KeyValueLogMessage.of("Cached meta-data is out-of-date",
-                                                               subspaceProvider.logKey(), subspaceProvider.toString(context),
-                                                               LogMessageKeys.VERSION, currentVersion,
-                                                               "cachedVersion", recordMetaData.getVersion()));
+                                                subspaceProvider.logKey()     , subspaceProvider.toString(context),
+                                                LogMessageKeys.VERSION        , currentVersion,
+                                                LogMessageKeys.CACHED_VERSION , recordMetaData.getVersion()));
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java]: Remove column alignment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java [hunk 6]
```diff
@@ -2431,11 +2431,11 @@ private boolean areAllRecordTypesSince(@Nullable Collection<RecordType> recordTy
...
-                    LogMessageKeys.INDEX_NAME, index.getName(),
-                    "indexVersion", index.getLastModifiedVersion(),
-                    "reason", reason.name(),
-                    subspaceProvider.logKey(), subspaceProvider.toString(context),
-                    LogMessageKeys.SUBSPACE_KEY, index.getSubspaceKey());
+                                            LogMessageKeys.INDEX_NAME    , index.getName(),
+                                            LogMessageKeys.INDEX_VERSION , index.getLastModifiedVersion(),
+                                            LogMessageKeys.REASON        , reason.name(),
+                                            subspaceProvider.logKey()    , subspaceProvider.toString(context),
+                                            LogMessageKeys.SUBSPACE_KEY  , index.getSubspaceKey());
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java]: Remove column alignment.  I'll stop adding more such comments from here on.. :)

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java [hunk 7]
```diff
@@ -32,14 +32,28 @@
...
     CALLING_CLASS("calling_class"),
     CALLING_METHOD("calling_method"),
     CALLING_LINE("calling_line"),
     FUTURE_COMPLETED("future_completed"),
     SCAN_PROPERTIES("scan_properties"),
     READ_VERSION("read_version"),
+    OLD("old"),
+    NEW("new"),
```
Comment (scgray) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: I'm kind of on the fence about adding such constants that are used by test cases or, generally, unlikely to ever appear in other code.  Not sure if anyone else has a strong opinion of this.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java (hunk 2 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java]: DONE
- Removed column alignment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java (hunk 4 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java]: DONE
- Removed column alignment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java (hunk 5 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java]: DONE
- Removed column alignment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java (hunk 6 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java]: DONE
- Removed column alignment everywhere it was added in previous commit.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java (hunk 1 above)
Comment (alecgrieser) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: I suppose the concern would be if someone were using the log names as part of their monitoring routine. (Like, if they had scripts or something that searched through logs for key/value pairs with certain names.)

I think it's still worth it to normalize on underscores, but that's the risk.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java [hunk 8]
```diff
@@ -65,12 +80,14 @@
     EXPECTED_COLUMN_SIZE("expected_column_size"),
     ACTUAL_COLUMN_SIZE("actual_column_size"),
     KEY_EXPRESSION("key_expression"),
+    KEY_EXPRESSION_C("keyExpression"),
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: I guess in line with https://github.com/FoundationDB/fdb-record-layer/pull/611/files#r287116009, I think the `_C` expressions should be converted where used to the under_score version of the key, especially as fixing that later would require removing keys from an enum, which is technically a breaking change.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java (hunk 7 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: - These keys are used in 'fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java'. The key names are not idea. This is not test code.
- It was decided to not standardize key names at this point.
- Standardizing at this point would be a trivial exercise though.

Comment (alecgrieser) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: It does seem a little weird to add test-case only keys (like, say, `TOTAL_RESULT_COUNT`). In theory, those could go in their own enum? `TestLogMessageKeys`?

Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: DONE:
- Moved all test-only keys to TestLogMessageKeys.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java (hunk 8 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: DONE
- Normalized all keys to the underscore variant.
- Does this warrant an update to ReleaseNotes?

**Comment by alecgrieser** [comment]
Looks like there is a unit test that expects camelCase:

```
com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTest.open() > FAILED

org.opentest4j.AssertionFailedError: Expected additional logging info name 'localVersion', but it is not present
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:36)
	at org.junit.jupiter.api.Assertions.fail(Assertions.java:62)
	at com.apple.foundationdb.record.TestHelpers.assertThrows(TestHelpers.java:185)
	at com.apple.foundationdb.record.TestHelpers.assertThrows(TestHelpers.java:154)
	at com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTest.open(FDBRecordStoreTest.java:1889)
```

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java (hunk 1 above)
Comment (vsharma999) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/LogMessageKeys.java]: Checked with concerned parties and found no known dependency on existing log keys from fdb-record-layer. After consultation, taking the risk.

**Pull request merged**